qiskit/circuit/library/standard_gates/xx_minus_yy.py

```diff
@@ -177,3 +177,13 @@ def power(self, exponent: float):
         """Raise gate to a power."""
         theta, beta = self.params
         return XXMinusYYGate(exponent * theta, beta)
+
+    def __eq__(self, other):
+        if isinstance(other, XXMinusYYGate):
+            try:
+                return math.isclose(
+                    self.params[0], other.params[0], rel_tol=0, abs_tol=1e-10
+                ) and math.isclose(self.params[1], other.params[1], rel_tol=0, abs_tol=1e-10)
+            except TypeError:
+                return self.params[0] == other.params[0] and self.params[1] == other.params[1]
```

Comment:
(Took me this many files to realise this, but here goes:)

This single `try/except` structure implies that if there's a mix of `ParameterExpression` and `float` params within a single gate instance, we'll compare the floats in exact equality, but if there's _no_ `ParameterExpression` we'll compare with fuzzy matching.  It's a bit of an edge case, but do we potentially want to reorganise the logic to ensure that the `try/catch` is per scalar?

That'd certainly be less tedious to do if we do decide to separate the scalar angle-like comparison into its own shared function, and probably not worth worrying about if we don't do that.

Reply:
Oh, yeah that's a good point. I was trying to avoid a loop for these, but yeah it completely falls apart with mixed `Parameter`s and numeric values.  We need to do what I did in https://github.com/Qiskit/qiskit/pull/11278 for gates with multiple parameters to do the try element-wise.

Reply:
Fixed in: https://github.com/Qiskit/qiskit/pull/11370/commits/d46cf05cc6011d16378e911c113c4f388177f9d0